instrumentation/jdbc/src/test/java/com/splunk/opentelemetry/instrumentation/jdbc/sqlserver/SqlServerTest.java

```diff
@@ -0,0 +1,366 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.instrumentation.jdbc.sqlserver;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.javaagent.bootstrap.CallDepth;
+import java.nio.charset.StandardCharsets;
+import java.sql.CallableStatement;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.MSSQLServerContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+
+class SqlServerTest {
+  private static final Logger logger = LoggerFactory.getLogger(SqlServerTest.class);
+
+  @RegisterExtension
+  static final AgentInstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  private static final MSSQLServerContainer sqlServer =
+      new MSSQLServerContainer("mcr.microsoft.com/mssql/server:2022-latest").acceptLicense();
+  private static Connection connection;
+
+  static {
+    sqlServer.withLogConsumer(new Slf4jLogConsumer(logger));
+  }
+
+  @BeforeAll
+  static void setup() throws Exception {
+    sqlServer.start();
+    connection = DriverManager.getConnection(sqlServer.getJdbcUrl(), "sa", sqlServer.getPassword());
+
+    Statement createTable = connection.createStatement();
+    createTable.execute("CREATE TABLE test_table (value INT NOT NULL)");
+    createTable.close();
+
+    testing.waitForTraces(1);
+    testing.clearData();
+  }
+
+  @AfterAll
+  static void cleanup() throws Exception {
+    connection.close();
```

Comment:
It looks like the tests create their own connection. Is there a reason to hold it open for the duration? Would it be bad to close the connection after creating the table in the `@BeforeAll`?